trends_presentation/src/commonMain/kotlin/net/thechance/mena/trends/presentation/screen/upload_reel/UploadReelScreen.kt

```diff
@@ -110,47 +109,45 @@ private fun UploadReelScreenContent(
                 .verticalScroll(rememberScrollState())
                 .padding(horizontal = Theme.spacing._16)
         ) {
-            UploadReelScreenHeader(
-                modifier = Modifier
-                    .padding(
-                        top = Theme.spacing._16,
-                        bottom = Theme.spacing._24
-                    )
-            )
+            UploadReelScreenHeader(Modifier.padding(top = Theme.spacing._16))
 
             UploadVideoCard(
+                modifier = Modifier.padding(top = Theme.spacing._24),
                 thumbnail = state.thumbnail,
                 isEnabled = state.isUploadVideoCardEnabled,
                 onCardClick = launcher::launch,
-                onEditClick = {
-                    listener.onEditVideoClick()
-                    launcher.launch()
-                },
-                modifier = Modifier.padding(bottom = Theme.spacing._24)
+                onEditClick = launcher::launch
             )
 
-            VideoLoadingCardItem(
-                title = "${state.selectedFile.name}.${state.selectedFile.extension}",
-                videoSize = state.selectedFile.sizeInMegaBytes,
-                videoState = state.uploadingTrendState,
-                progress = state.uploadedBytes.toFloat() / state.selectedFile.sizeInBytes.toFloat(),
-                modifier = Modifier.padding(bottom = Theme.spacing._16),
-                onAction = { action ->
-                    when (action) {
-                        VideoAction.Cancel -> listener.onCancelUploadClick()
-                        VideoAction.Retry -> listener.onRetryUploadClick()
-                        VideoAction.Delete -> listener.onDeleteVideoClick()
+            if (state.uploadingState != UploadReelScreenState.UploadingReelState.IDLE) {
+                VideoLoadingCardItem(
+                    modifier = Modifier.padding(
+                        top = state.thumbnail?.let {
+                            Theme.spacing._24
+                        } ?: Theme.spacing._8
+                    ),
+                    title = state.selectedFile.name,
+                    sizeUploaded = state.sizeUploaded,
+                    videoSize = state.selectedFile.sizeText,
+                    videoState = state.uploadingState,
+                    progress = state.uploadingProgress,
+                    onAction = { action ->
+                        when (action) {
+                            VideoAction.Cancel -> listener.onCancelUploadClick()
+                            VideoAction.Retry -> listener.onRetryUploadClick()
+                            VideoAction.Delete -> listener.onDeleteVideoClick()
+                        }
```

Comment:
can we did this for readablty ?
or it doesnt worth it 
what do u think 

```suggestion 
  onAction = { action -> selectActon(action,listener) }
  
  // in the end of the file 
  
  private fun selectActon(action: VideoAction, listener: UploadReelInteractionListener){
    when (action) {
        VideoAction.Cancel -> listener.onCancelUploadClick()
        VideoAction.Retry -> listener.onRetryUploadClick()
        VideoAction.Delete -> listener.onDeleteVideoClick()
    }
}

Reply:
really, don't know if we need to do that
but we can use it like this:
```
    ....
    videoSize = state.selectedFile.sizeText,
    uploadingState = state.uploadingState,
    progress = state.uploadingProgress,
) { action ->
      when (action) {
           VideoAction.Cancel -> listener.onCancelUploadClick()
           VideoAction.Retry -> listener.onRetryUploadClick()
           VideoAction.Delete -> listener.onDeleteVideoClick()
     }
}
```
@iZyadNasser what do you think?